packages/regl-worldview/src/commands/GLTFScene.js

```diff
@@ -0,0 +1,210 @@
+
+  void main() {
+    vec3 baseColor = texture2D(baseColorTexture, vTexCoord).rgb;
+    float diffuse = light.diffuseIntensity * max(0.0, dot(vNormal, -light.direction));
+    gl_FragColor = vec4((light.ambientIntensity + diffuse) * baseColor, 1);
+  }
+  `,
+  });
+
+  let drawCalls;
+  function prepareDrawCallsIfNeeded(model) {
+    if (drawCalls) {
+      return;
+    }
+
+    const textures = model.json.textures.map((texture) => {
+      const sampler = model.json.samplers[texture.sampler];
+      return regl.texture({
+        data: model.images[texture.source],
+        min: "linear", //glConstantToRegl(sampler.minFilter),
+        mag: glConstantToRegl(sampler.magFilter),
+        wrapS: glConstantToRegl(sampler.wrapS),
+        wrapT: glConstantToRegl(sampler.wrapT),
+      });
+    });
+    drawCalls = [];
+
+    function drawMesh(mesh, nodeMatrix) {
+      for (const primitive of mesh.primitives) {
+        const material = model.json.materials[primitive.material];
+        const texInfo = material.pbrMetallicRoughness.baseColorTexture;
+        drawCalls.push({
+          indices: model.accessors[primitive.indices],
+          positions: model.accessors[primitive.attributes.POSITION],
+          normals: model.accessors[primitive.attributes.NORMAL],
+          texCoords: model.accessors[primitive.attributes[`TEXCOORD_${texInfo.texCoord || 0}`]],
+          baseColorTexture: textures[texInfo.index],
+          nodeMatrix,
+        });
+      }
+    }
+    function drawNode(node, parentMatrix) {
+      const nodeMatrix = node.matrix
+        ? mat4.clone(node.matrix)
+        : mat4.fromRotationTranslationScale(
+            mat4.create(),
+            node.rotation || [0, 0, 0, 1],
+            node.translation || [0, 0, 0],
+            node.scale || [1, 1, 1]
+          );
+      mat4.mul(nodeMatrix, parentMatrix, nodeMatrix);
+      if (node.mesh != null) {
+        drawMesh(model.json.meshes[node.mesh], nodeMatrix);
+      }
+      if (node.children) {
+        for (const childIdx of node.children) {
+          drawNode(model.json.nodes[childIdx], nodeMatrix);
+        }
+      }
+    }
+    for (const nodeIdx of model.json.scenes[model.json.scene].nodes) {
+      const rootTransform = mat4.create();
+      mat4.rotateX(rootTransform, rootTransform, Math.PI / 2);
+      mat4.rotateY(rootTransform, rootTransform, Math.PI / 2);
+      drawNode(model.json.nodes[nodeIdx], rootTransform);
+    }
+  }
+
+  const withPoseMatrix = regl({
+    context: {
+      poseMatrix: (context, props) =>
+        mat4.fromRotationTranslationScale(
+          mat4.create(),
+          orientationToVec4(props.pose.orientation),
+          pointToVec3(props.pose.position),
+          props.scale ? pointToVec3(props.scale) : [1, 1, 1]
+        ),
+    },
+  });
+
+  return (props) => {
+    prepareDrawCallsIfNeeded(props.model);
+    withPoseMatrix(props, () => {
+      command(drawCalls);
+    });
+  };
+};
+
+type Props = {|
+  modelURL: string,
```

Comment:
Hm why not make this part of `children` too? That seems to be the way that we do things in other commands?

Reply:
I kept this as a separate prop now that it can be either `string` or `() => Promise<Object>`.